test-common/utils.c

```diff
@@ -2104,125 +2104,307 @@ format_flatpak_ref_name_with_branch_override_arch (const gchar *name,
+          case FLATPAK_INSTALL_INFO_TYPE_EXTENSION:
+            {
+              /* If we weren't going to preinstall the runtime,
+               * uninstall it now */
+              if (!install_info->preinstall)
+                {
+                  if (!flatpak_uninstall (updater_dir,
+                                          formatted_ref_name,
+                                          error))
+                    return FALSE;
+                }
+            }
+            break;
+          case FLATPAK_INSTALL_INFO_TYPE_APP:
+            break;
+          default:
+            g_assert_not_reached();
+            return FALSE;
+        }
+
+      if (install_info->preinstall)
+        {
+          if (!flatpak_install (updater_dir,
+                                install_info->repo_name,
+                                formatted_ref_name,
+                                error))
+            return FALSE;
+        }
+    }
+
+  /* Now that we have finished preinstalling all the flatpaks,
+   * set the collection-id on all remote configs in the installation
+   * directory */
+  g_hash_table_iter_init (&iter, repository_infos);
+  while (g_hash_table_iter_next (&iter, &key, &value))
+    {
+      FlatpakRepoInfo *repo_info = value;
+
+      if (repo_info->remote_collection_id == NULL)
+        continue;
+
+      if (set_flatpak_remote_collection_id (updater_dir,
+                                            key,
+                                            repo_info->remote_collection_id,
+                                            error))
         return FALSE;
     }
 
   return TRUE;
 }
 
+gboolean
+eos_test_setup_flatpak_repo_with_preinstalled_apps_simple (GFile        *updater_dir,
+                                                           const gchar  *branch,
+                                                           const gchar  *repo_name,
+                                                           const gchar  *repo_collection_id,
+                                                           const gchar  *remote_config_collection_id,
+                                                           const gchar **flatpak_names,
+                                                           const gchar **preinstall_flatpak_names,
+                                                           GError      **error)
+{
+  g_autoptr(GHashTable) repo_infos = g_hash_table_new_full (g_str_hash,
+                                                            g_str_equal,
+                                                            g_free,
+                                                            (GDestroyNotify) flatpak_repo_info_free);
+  g_autoptr(GPtrArray) flatpak_install_infos = g_ptr_array_new_full (g_strv_length ((GStrv) flatpak_names) + 1,
+                                                                     (GDestroyNotify) flatpak_install_info_free);
+  const gchar **flatpak_names_iter = flatpak_names;
```

Comment:
Nitpick: Move this assignment down into the `for` loop, since it isn’t needed earlier, and it’s more readable to keep all the loop logic together